components/src/core/drawer/drawer-nav-item.tsx

```diff
@@ -14,6 +14,7 @@ export type NavItem = {
     icon?: any;
     itemID: string;
     items?: NestedNavItem[];
+    rightComponent?: JSX.Element;
```

Comment:
`rightComponent` is already a prop on the InfoListItem. We should be able to spread this without explicitly adding a prop for it.

Reply:
The spread on line 113 should already be overriding the rightComponent if it's provided. Curious why this is not working as expected.